lib/YaST/Firewall/MainPage.pm

```diff
@@ -0,0 +1,68 @@
+# SUSE's openQA tests
+#
+# Copyright 2021 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: The class introduces all accessing methods for YaST module
+# System Settings Page.
+# Maintainer: QE YaST <qa-sle-yast@suse.de>
+
+package YaST::Firewall::MainPage;
+use strict;
+use warnings;
+use testapi;
+
+sub new {
+    my ($class, $args) = @_;
+    my $self = bless {
+        app => $args->{app}
+    }, $class;
+    return $self->init();
+}
+
+sub init {
+    my $self = shift;
+    $self->{overview_tree} = $self->{app}->tree({id => "\"Y2Firewall::Widgets::OverviewTree\""});
+    $self->{btn_accept} = $self->{app}->button({id => 'next'});
+    return $self;
+}
+
+sub is_shown {
+    my ($self) = @_;
+    my $is_shown = $self->{overview_tree}->exist();
+    save_screenshot if $is_shown;
+    return $is_shown;
+
+}
+
+sub select_start_up_page {
+    my ($self) = @_;
+    $self->{overview_tree}->select("Start-Up");
+    save_screenshot;
+}
+
+sub select_interfaces_page {
+    my ($self) = @_;
+    $self->{overview_tree}->select("Interfaces");
+    save_screenshot;
+}
+
+sub select_zones_page {
+    my ($self) = @_;
+    $self->{overview_tree}->select("Zones");
+    save_screenshot;
+}
+
+sub press_accept {
```

Comment:
you mixed here this idea of main page with the specific page.
We probably need to extract here the bottom buttons and the tree and call it for example 'BasePage' . Because 'main' doesn't have that meaning.
then the specific pages will inherit this basepage.

Reply:
The other page do not need any action of main page's button(accept button etc..), so from my view no need BasePage in our case